### PR TITLE
Fix memory leaks in EEID plugins and tests.

### LINKED ISSUE #3368: EEID_PLUGIN test failed, having memory leak
eeid_plugin test failed for both master and feature/sgxlkl-support branch on ACC VM. May be a potential regression issue.

Here is the error output:

```
Start 254: tests/eeid_plugin
1/1 Test #254: tests/eeid_plugin ................Child aborted***Exception:   8.13 sec

0% tests passed, 1 tests failed out of 1

Total Test time (real) =   8.14 sec

The following tests FAILED:
        254 - tests/eeid_plugin (Child aborted)
Errors while running CTest
```

### PR DESCRIPTION
**Pull request by wintersteiger**
This fixes memory leaks in the EEID code. 

Fixes #3368.

Signed-off-by: Christoph M. Wintersteiger <cwinter@microsoft.com>

### CODE REVIEW

File: tests/eeid_plugin/test_helpers.c [hunk 1]
```diff
@@ -24,7 +24,7 @@ oe_result_t make_test_eeid(oe_eeid_t** eeid, size_t data_size)
     for (size_t i = 0; i < data_size; i++)
         (*eeid)->data[i] = 'a' + (i % 26);
     (*eeid)->data[data_size - 1] = 0;
-    (*eeid)->size_settings.num_heap_pages = 100 + (data_size / OE_PAGE_SIZE);
+    (*eeid)->size_settings.num_heap_pages = 110 + (data_size / OE_PAGE_SIZE);
```
Comment (jxyang) [inline tests/eeid_plugin/test_helpers.c:27]: Why is this needed?

**Comment by jxyang** [comment]
These are genuine memory leaks. Now the interesting question is why the pipeline runs (with EEID enabled) never report failures, and now the memory check kicks in with make but not ninja.

File: tests/eeid_plugin/test_helpers.c (hunk 1 above)
Comment (wintersteiger) [reply tests/eeid_plugin/test_helpers.c:27]: For some reason the latest master needs a bit more memory, I didn't investigate it any deeper, but without that it ran out of heap space.

**Comment by wintersteiger** [comment]
@jxyang Yeah, this is a mystery to me too. I assumed this would work the same everywhere, but apparently it doesn't.

**Comment by anakrish** [comment]
bors try

**Comment by wintersteiger** [comment]
bors try

File: common/sgx/eeid_verifier.c [hunk 2]
```diff
@@ -393,7 +394,12 @@ static oe_result_t _eeid_free_claims_list(
 {
     OE_UNUSED(context);
     OE_UNUSED(claims_size);
-    free(claims);
+    for (size_t i = 0; i < claims_size; i++)
```
Comment (CodeMonkeyLeet) [inline common/sgx/eeid_verifier.c:397]: Consider making `sgx_attestation_plugin_free_claims_list` part of include/openenclave/internal/sgx/plugin.h instead and using that here.

File: common/sgx/eeid_verifier.c [hunk 3]
```diff
@@ -324,6 +323,8 @@ static oe_result_t _eeid_verify_evidence(
         uint64_t r_attributes = parsed_report.identity.attributes;
         uint32_t r_id_version = parsed_report.identity.id_version;
 
+        oe_free_claims(sgx_claims, sgx_claims_length);
```
[the remark this answers is not in the document]
Comment (wintersteiger) [reply common/sgx/eeid_verifier.c:326]: Oh yes, that's not intentional at all. Check added!

File: common/sgx/eeid_verifier.c (hunk 2 above)
Comment (wintersteiger) [reply common/sgx/eeid_verifier.c:396]: That's a good suggestion; done. I kept the name as suggested in the design doc, but it's a bit long.

**Comment by wintersteiger** [comment]
Thanks @CodeMonkeyLeet for the review! You're right, there is a bunch of duplicated code from the SGX plugin. What I really want is to call the SGX plugin from the EEID plugin to avoid that.

Regarding `memalign`: Yes, this is broken, I'll pull the relevant changes from my Windows branch over into this one.

File: common/sgx/eeid_verifier.c (hunk 2 above)
Comment (wintersteiger) [reply common/sgx/eeid_verifier.c]: Oh yeah, thanks for checking, I forgot that end of the story :-) Done now.

**Comment by wintersteiger** [comment]
bors try

**Comment by wintersteiger** [comment]
I think this one's ready to merge, except that the successful bors status wasn't picked up yet. I've seen that happen in other PRs, is there are more general problem with bors and/or its github integration?

**Comment by wintersteiger** [comment]
bors try

**Comment by wintersteiger** [comment]
bors try

**Pull request merged**